src/main/java/net/minestom/server/extras/lan/OpenToLAN.java

```diff
@@ -0,0 +1,192 @@
+ */
+public class OpenToLAN {
+    private static final InetSocketAddress PING_ADDRESS = new InetSocketAddress("224.0.2.60", 4445);
+    private static final String PING_FORMAT = "[MOTD]%s[/MOTD][AD]%s[/AD]";
+
+    private static final LegacyComponentSerializer LEGACY_SERIALIZER = LegacyComponentSerializer.legacySection();
+    private static final Logger LOGGER = LoggerFactory.getLogger(OpenToLAN.class);
+
+    private static volatile Component description = Component.text("A Minestom server", NamedTextColor.AQUA);
+    private static volatile DatagramPacket packet = null;
+
+    private static volatile DatagramSocket socket = null;
+    private static volatile Task task = null;
+
+    private OpenToLAN() { }
+
+    /**
+     * Opens the server to LAN, printing a warning message if an error occurred.
+     *
+     * @return {@code true} if it was opened, {@code false} if it was already or couldn't be opened
+     * @see OpenToLAN
+     */
+    public static boolean open() {
+        try {
+            return openWithException();
+        } catch (IOException e) {
+            LOGGER.warn("Could not open a socket for LAN connections!", e);
+            return false;
+        }
+    }
+
+    /**
+     * Opens the server to LAN, printing a warning message if an error occurred.
+     *
+     * @param port the port to send the LAN packets from
+     * @return {@code true} if it was opened, {@code false} if it was already or couldn't be opened
+     * @see OpenToLAN
+     */
+    public static boolean open(int port) {
+        try {
+            return openWithException(port);
+        } catch (SocketException e) {
+            LOGGER.warn("Could not open a socket for LAN connections!", e);
+            return false;
+        }
+    }
+
+    /**
+     * Opens the server to LAN.
+     *
+     * @return {@code true} if it was opened, {@code false} if it was already opened
+     * @throws IOException if an error occurred
+     * @see OpenToLAN
+     */
+    public static boolean openWithException() throws IOException {
+        if (socket != null) {
+            return false;
+        } else {
+            ServerSocket socket = new ServerSocket(0);
+            int port = socket.getLocalPort();
+            socket.close();
+
+            return open(port);
+        }
+    }
+
+    /**
+     * Opens the server to LAN.
+     *
+     * @param port the port to send the LAN packets from
+     * @return {@code true} if it was opened, {@code false} if it was already opened
+     * @throws SocketException if an error occurred
+     * @see OpenToLAN
+     */
+    public static boolean openWithException(int port) throws SocketException {
+        if (socket != null) {
+            return false;
+        } else {
+            socket = new DatagramSocket(port);
+            task = MinecraftServer.getSchedulerManager().buildTask(OpenToLAN::ping)
+                    .repeat(15, TimeUnit.SECOND)
```

Comment:
The wikivg page says 1.5, not 15

Reply:
jesus... maybe i should make this configurable lmao